src/analysis/normed_space/multilinear.lean

```diff
@@ -0,0 +1,869 @@
+let f' : multilinear_map 𝕜 (λ(i : fin n), E i.succ) (E 0 →L[𝕜] E₂) :=
+{ to_fun := λm, (f.to_multilinear_map.curry_right m).mk_continuous
+    (∥f∥ * univ.prod (λ(i : fin n), ∥m i∥)) $ λx, begin
+      change ∥f (cons x m)∥ ≤ ∥f∥ * finset.prod univ (λ (i : fin n), ∥m i∥) * ∥x∥,
+      rw [mul_assoc, mul_comm _ (∥x∥), ← mul_assoc],
+      exact f.norm_map_cons_le x m,
+    end,
+  add  := λ m i x y, by { simp, refl },
+  smul := λ m i c x, by { simp, refl } } in
+f'.mk_continuous (∥f∥) (λm, linear_map.mk_continuous_norm_le _
+  (mul_nonneg' (norm_nonneg _) (prod_nonneg (λj hj, norm_nonneg _))) _)
+
+@[simp] lemma continuous_multilinear_map.curry_right_apply
+  (f : continuous_multilinear_map 𝕜 E E₂) (x : E 0) (m : Π(i : fin n), E i.succ) :
+  f.curry_right m x = f (cons x m) := rfl
+
+@[simp] lemma continuous_multilinear_map.curry_uncurry_right
+  (f : continuous_multilinear_map 𝕜 (λ(i : fin n), E i.succ) (E 0 →L[𝕜] E₂)) :
+  f.uncurry_right.curry_right = f :=
+begin
+  ext m x,
+  simp only [cons_zero, continuous_multilinear_map.curry_right_apply,
+             continuous_multilinear_map.uncurry_right_apply],
+  rw tail_cons
+end
+
+@[simp] lemma continuous_multilinear_map.uncurry_curry_right
+  (f : continuous_multilinear_map 𝕜 E E₂) : f.curry_right.uncurry_right = f :=
+by { ext m, simp }
+
+@[simp] lemma continuous_multilinear_map.curry_right_norm
+  (f : continuous_multilinear_map 𝕜 E E₂) : ∥f.curry_right∥ = ∥f∥ :=
+begin
+  refine le_antisymm (multilinear_map.mk_continuous_norm_le _ (norm_nonneg _) _) _,
+  have : ∥f.curry_right.uncurry_right∥ ≤ ∥f.curry_right∥ :=
+    multilinear_map.mk_continuous_norm_le _ (norm_nonneg _) _,
+  simpa
+end
+
+@[simp] lemma continuous_multilinear_map.uncurry_right_norm
+  (f : continuous_multilinear_map 𝕜 (λ(i : fin n), E i.succ) (E 0 →L[𝕜] E₂)) :
+  ∥f.uncurry_right∥ = ∥f∥ :=
+begin
+  refine le_antisymm (multilinear_map.mk_continuous_norm_le _ (norm_nonneg _) _) _,
+  have : ∥f.uncurry_right.curry_right∥ ≤ ∥f.uncurry_right∥ :=
+    multilinear_map.mk_continuous_norm_le _ (norm_nonneg _) _,
+  simpa
+end
+
+variables (𝕜 E E₂)
+
+/-- The space of continuous multilinear maps on `Π(i : fin (n+1)), E i` is canonically isomorphic to
+the space of continuous multilinear maps on `Π(i : fin n), E i.succ` with values in the space of
+continuous linear maps on `E 0`, by separating the first variable. We register this isomorphism as a
+linear isomorphism in `continuous_multilinear_curry_right_equiv_aux 𝕜 E E₂`.
+The algebraic version (without continuity assumption on the maps) is
+`multilinear_curry_right_equiv 𝕜 E E₂`, and the topological isomorphism (registering
+additionally that the isomorphism is continuous) is
+`continuous_multilinear_curry_right_equiv 𝕜 E E₂`.
+
+The direct and inverse maps are given by `f.uncurry_right` and `f.curry_right`. Use these
+unless you need the full framework of linear equivs. -/
+def continuous_multilinear_curry_right_equiv_aux :
```

Comment:
Is there any specific reason for introducing `*_equiv_aux`? If you're going to use it outside of this file, then it probably deserves a better name. If not, then why not merge with `*_equiv`?

Reply:
This was my first move, but I was told on Zulip that it is not a problem to have auxiliary definitions like that, and that it makes it easier for Lean to swallow.